app/controllers/api/users.php

```diff
@@ -63,7 +63,7 @@ function createUser(string $hash, mixed $hashOptions, string $userId, ?string $e
             'status' => true,
             'password' => (!empty($password)) ? ($hash === 'plaintext' ? Auth::passwordHash($password, $hash, $hashOptionsObject) : $password) : null,
             'hash' => $hash === 'plaintext' ? Auth::DEFAULT_ALGO : $hash,
-            'hashOptions' => $hash === 'plaintext' ? Auth::DEFAULT_ALGO_OPTIONS : $hashOptions,
+            'hashOptions' => $hash === 'plaintext' ? Auth::DEFAULT_ALGO_OPTIONS : $hashOptionsObject + ['type' => $hash],
```

Comment:
Why do we need type in hashOptions? the `hash` is the type

Reply:
Also I think we need to apply this to account controller, and teams controller.

Reply:
After reading more of the PR I see why we do this. But this now requires migration.

Reply:
I am making myself note to double check migrations between 1.2.x and master